cate/ds/esa_cci_odp.py

```diff
@@ -741,45 +733,54 @@ async def _create_data_source(self, json_dict: dict, datasource_id: str):
                 meta_info.update(json_dict)
-                self._adjust_json_dict(meta_info, value_tuple)
+                self._adjust_json_dict(meta_info, drs_id)
                 meta_info['cci_project'] = meta_info['ecv']
-                data_source = EsaCciOdpDataSource(self, meta_info, datasource_id, pretty_id, value_tuple)
+                meta_info['fid'] = datasource_id
+                data_source = EsaCciOdpDataSource(self, meta_info, datasource_id, drs_id)
                 self._data_sources.append(data_source)
 
-    def _adjust_json_dict(self, json_dict: dict, value_tuple: tuple):
-        self._adjust_json_dict_for_param(json_dict, 'time_frequency', 'time_frequencies', value_tuple[0])
-        self._adjust_json_dict_for_param(json_dict, 'processing_level', 'processing_levels', value_tuple[1])
-        self._adjust_json_dict_for_param(json_dict, 'data_type', 'data_types', value_tuple[2])
-        self._adjust_json_dict_for_param(json_dict, 'sensor_id', 'sensor_ids', value_tuple[3])
-        self._adjust_json_dict_for_param(json_dict, 'platform_id', 'platform_ids', value_tuple[4])
-        self._adjust_json_dict_for_param(json_dict, 'product_string', 'product_strings', value_tuple[5])
-        self._adjust_json_dict_for_param(json_dict, 'product_version', 'product_versions', value_tuple[6])
+    def _adjust_json_dict(self, json_dict: dict, drs_id: str):
+        values = drs_id.split('.')
+        self._adjust_json_dict_for_param(json_dict, 'time_frequency', 'time_frequencies',
+                                         self._convert_time_from_drs_id(values[2]))
+        self._adjust_json_dict_for_param(json_dict, 'processing_level', 'processing_levels', values[3])
+        self._adjust_json_dict_for_param(json_dict, 'data_type', 'data_types', values[4])
+        self._adjust_json_dict_for_param(json_dict, 'sensor_id', 'sensor_ids', values[5])
+        self._adjust_json_dict_for_param(json_dict, 'platform_id', 'platform_ids', values[6])
+        self._adjust_json_dict_for_param(json_dict, 'product_string', 'product_strings', values[7])
+        self._adjust_json_dict_for_param(json_dict, 'product_version', 'product_versions', values[8])
+
+    def _convert_time_from_drs_id(self, time_value: str) -> str:
+        if time_value == 'mon' or time_value == 'month':
+            return 'month'
+        if time_value == 'yr' or time_value == 'year':
+            return 'year'
+        if time_value == 'day':
+            return 'day'
+        if time_value == 'satellite-orbit-frequency':
+            return 'satellite-orbit-frequency'
+        if time_value == '5-days':
+            return '5 days'
+        if time_value == '8-days':
+            return '8 days'
+        if time_value == '15-days':
+            return '15 days'
+        if time_value == '13-yrs':
+            return '13 years'
+        if time_value == 'climatology':
+            return 'climatology'
+        raise ValueError('Unknown time frequency format')
```

Comment:
Convert the many `if`-s into simple lookup using a dict. 

B.t.w. why is this needed at all?

Reply:
1. Better readability. These will be part of the metdata
2. In this notation, they can be used in opensearch queries.